text/0060-linking-backend-errors-with-replays.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 2023-01-13
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/60
+- RFC Status: draft
+
+# Summary
+
+Today, replays are associated with frontend events (errors, transactions, etc.) via tagging the events with the replay_id. We also tag replays with the associated event_id, which allows us to search / rank replays by their associated errors.
+
+We'd like to take this a step further, and link downstream events that occur on the backend with the replay.
+
+This feature will likely require cross-team collaboration, so a few options are presented below.
+
+# Motivation
+
+Far and away the most useful replays for developers are those that occur with errors associated. There are cases where no frontend error occurs, but a backend one does. A developer seeing this backend error could find a replay that caused it extremely useful. For one, a developer can see how a backend error manifests in the user experience and how impactful/detrimental it was to the user experience. This bridges the backend to the frontend. Alternatively, if a developer is debugging a frontend error, they can more easily uncover if the root cause was something deeper in the stack (i.e., stemming from the backend).
+
+Also, we're able to give a more complete user session view by communicating to customers _all_ errors that happen in their user session.
+
+Additionally, part of our Session Replay strategy is to more deeply integrate replays in the existing product experiences to increase the overall value of the Sentry platform. Issues is our bread and butter, and we want to be able to provide valuable replays for frontend and backend errors.
+
+On the performance side, we want to be able to connect backend transactions to replays. For example, seeing how a slow API translates to the user experience.
+
+## Supporting quotes
+
+> XX does a good job of displaying the session replays. To me the biggest differentiator that Session Replay has is its connectivity to the rest of the Sentry ecosystem.
+> XX is perfect for determining front-end only issues, but doesn't give enough information to find root causes when the issue is deeper in the system. and It would be great if we could link replays to BE errors. This is our current setup. 1) FE Web App; 2) BE Web App.
+
+# Options Considered
+
+## Option A: Dynamic Sampling Context / Baggage Header
+
+[Link To relevant docs about dynamic sampling context / baggage](https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#baggage)
+
+We'd add a new baggage http header `sentry-replay_id`. This value would automatically make it onto the trace envelope header, and from there we could add it onto the event's payload if it exists.
```

Comment:
Just for clarification. There is only one `baggage` HTTP header and it contains multiple items. A example baggage header looks like this: `"other-vendor-value-1=foo;bar;baz, sentry-trace_id=771a43a4192642f0b136d5159a501700, sentry-public_key=49d0f7386ad645858ae85020e393bef3, sentry-sample_rate=0.01337, sentry-user_id=Am%C3%A9lie, other-vendor-value-2=foo;bar;"`

Reply:
right, makes sense. will clarify "we add a new baggage item"